### PR TITLE
fix: allow readable stream type in FromStream constructor

### PR DESCRIPTION
**Pull request by patrickmichalina**

### CODE REVIEW

**Comment by patrickmichalina** [comment]
Not exactly sure how you want to handle this but the current `FromStream` constructor is not quite right. Since I feed in a `Readable` stream and use a `Writable` stream on the output. In my app I currently have to cast a `Readable` input stream to `as any`.

File: lib/types.ts [hunk 1]
```diff
@@ -920,7 +920,9 @@ export class FromStream implements IOOperation {
         return this.ioId
     }
     async toOutput(buffer: ArrayBuffer, collector: Object): Promise<any> {
```
Comment (patrickmichalina) [inline lib/types.ts:922]: shouldn't this change from
`async toOutput(buffer: ArrayBuffer, collector: Object): Promise<any>`
to
`toOutput(buffer: ArrayBuffer, collector: Object): void`

**Comment by lilith** [comment]
@anshulrgoyal Could you look at this?

**Comment by anshulrgoyal** [comment]
ok

File: lib/types.ts (hunk 1 above)
Comment (anshulrgoyal) [reply lib/types.ts:922]: yeah.

**Review by anshulrgoyal — APPROVED** [review]
LGTM

**Comment by patrickmichalina** [comment]
My original motivation for submitting this PR was to fix mostly a typescript type issue. However after moving this closer to a production app I am actually seeing race conditions occur when working with google cloud storage. Relying on this promise is not working since the stream isn't complete and the resolve is not triggered correctly. My PR actually fixes the issue though :)

**Comment by anshulrgoyal** [comment]
@patrickmichalina The changes look good but can u add a simple test case in test and an example for this in the readme. It will be highly appreciated.

**Pull request merged**